chips/sam4l/src/acifc.rs

```diff
@@ -0,0 +1,370 @@
+//! Implementation of the GPIO controller.
+//!
+//! See datasheet section "37. Analog Comparator Interface (ACIFC)".
+//!
+//! The Analog Comparator Interface (ACIFC) controls a number of Analog Comparators (AC) with identical behavior. Each Analog Comparator compares two voltages and gives a compare output depending on this comparison.
+//! A specific AC is referred to as ACx where x is any number from 0 to n and n is the index of last AC module. The ACIFC on the SAM4L supports a total of 8 ACs (and therefore 4 possible ACWs).
+//! However, note that the 64 pin SAM4L (e.g. on the Hail) has 2 ACs (ACA0 and ACB0), and the 100 pin SAM4L (e.g. on the Imix) has 4 ACs (ACA0, ACB0, ACA1, ACB1). Currently, no version of the SAM4L exists with all the 8 ACs implemented.
+//! Therefore a lot of the defined bitfields remaind unused, but are initialized for a possible future scenario.
+//! The ACIFC can be configured in normal mode using each comparator independently or in window mode using defined comparator pairs (ACx and ACx+1) to observe a window.
+//!
+//! Author: Danilo Verhaert <verhaert@cs.stanford.edu>
+
+// TODO:
+// - Implement handling of interrupts
+// - Implement other modes, e.g. user and peripheral triggered comparison
+
+use kernel::ReturnCode;
+use kernel::common::regs::{ReadOnly, ReadWrite, WriteOnly};
+use kernel::hil;
+use pm;
+
+#[repr(C)]
+pub struct AcifcRegisters {
+    ctrl: ReadWrite<u32, Control::Register>,
+    sr: ReadOnly<u32, Status::Register>,
+    _reserved0: [ReadOnly<u32>; 2],
+    ier: WriteOnly<u32, Interrupt::Register>,
+    idr: WriteOnly<u32, Interrupt::Register>,
+    imr: ReadOnly<u32, Interrupt::Register>,
+    isr: ReadOnly<u32, Interrupt::Register>,
+    icr: WriteOnly<u32, Interrupt::Register>,
+    tr: ReadWrite<u32, Test::Register>,
+    _reserved1: [ReadOnly<u32>; 2],
+    parameter: ReadOnly<u32, Parameter::Register>,
+    version: ReadOnly<u32>,
+    _reserved2: [ReadOnly<u32>; 18],
+    confw: [ReadWrite<u32, WindowConfiguration::Register>; 4],
+    _reserved3: [ReadOnly<u32>; 16],
+    conf: [ReadWrite<u32, ACConfiguration::Register>; 8],
+}
+
+register_bitfields![u32,
+	Control [
+		// Analog comparator test mode. Equal to 1 means AC outputs will be bypassed with values in AC test register.

```

Comment:
If you make these doc comments (`///` instead of `//`) they'll show up in the auto-generated documentation